### DISCUSSION TITLE
TPCH PostgreSQL Table Partitionning

### DISCUSSION THREAD

**Discussion by SylvieEmpereur** [TPC-Council/HammerDB#369] [Q&A] [answered]
Hello everyone,

Nice to play with the many capabilities of hammerdb tool.  Thanks to the developers !

A question regarding table partitioning when using community PostgreSQL.  Is there support for partitioning of large tables in TPCH ?
Looking into code, it looks like compression / distribution of data is reserved for Greenplum.  Is that correct or did I miss a feature ?

Regards,
Sylvie

**Comment by sm-shaw** [comment, marked as answer]
That is correct, for now HammerDB does not partition the TPC-H schema for any database. This is because initially with the commerical databases partitioning is a paid option and not necessarily supported in all environments, therefore HammerDB built a schema that didn't rely on additional options.  Also the open source databases didn't support analytic features such as column store or partitioning. 

Therefore for now the way to partition the schema is to use the datagen option https://www.hammerdb.com/docs/ch13.html to create and load your own partitioned schema.  The best approach to partitioning TPC-H is to partition LINEITEM by date and there are a number of examples to do this. 
Now PostgreSQL supports parallel query, It would be a viable issue to add the feature to partition lineitem for whenever this could be done.

**Comment by SylvieEmpereur** [comment]
Hi Steve,

Thank you very much for your quick response.

Best Regards,
Sylvie
